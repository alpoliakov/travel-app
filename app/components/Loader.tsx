import React from 'react';

export default function Loader({ show }) {
  return (
    show && (
      <div className="w-full h-full fixed block top-0 left-0 bg-gray-700 opacity-75 z-50">
        <div
          style={{ top: '50%' }}
          className="loader top-1/2 my-0 mx-auto block relative w-0 h-0"></div>
      </div>
    )
  );
}
