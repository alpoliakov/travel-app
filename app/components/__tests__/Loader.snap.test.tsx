import React from 'react';
import renderer from 'react-test-renderer';

import Loader from '../Loader';

describe('Loader render correctly', () => {
  test('Loader show - true', () => {
    const tree = renderer.create(<Loader show={true} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Loader show - false', () => {
    const tree = renderer.create(<Loader show={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
