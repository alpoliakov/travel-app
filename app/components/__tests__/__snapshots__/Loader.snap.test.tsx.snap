// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader render correctly Loader show - false 1`] = `null`;

exports[`Loader render correctly Loader show - true 1`] = `
<div
  className="w-full h-full fixed block top-0 left-0 bg-gray-700 opacity-75 z-50"
>
  <div
    className="loader top-1/2 my-0 mx-auto block relative w-0 h-0"
    style={
      Object {
        "top": "50%",
      }
    }
  />
</div>
`;
